apps/src/code-studio/progressRedux.ts

```diff
@@ -313,6 +314,35 @@ export function navigateToLevelId(levelId: string): ProgressThunkAction {
   };
 }
 
+// Updates the current level ID in the store when the level
+// (and possibly sublevel) index changes.
+// Typically happens when the user presses the browser back/forward button
+// in a level progression that doesn't require page reloads.
+export function onLevelIndexChange(
+  levelIndex: number,
+  sublevelIndex?: number
+): ProgressThunkAction {
+  return (dispatch, getState) => {
+    const levels: LevelWithProgress[] = getCurrentLevels(getState());
+    if (!levels || levels.length === 0) {
+      return;
+    }
+
+    const level = levels[levelIndex];
+    let newLevelId = level.id;
+    if (
+      sublevelIndex !== undefined &&
+      level.sublevels &&
+      sublevelIndex < level.sublevels.length
+    ) {
+      const sublevel = level.sublevels[sublevelIndex];
+      newLevelId = sublevel.id;
+    }
+
+    dispatch(setCurrentLevelId(newLevelId));
```

Comment:
I generally try to avoid using the `let` keyword. It adds some mental overhead for reviewers and future readers of the code trying to understand that the variable actually had something assigned to it in every case where it should, and trying to understand where the variable is reassigned.

In this case avoiding all that seems easy enough.

```suggestion
    if (
      sublevelIndex !== undefined &&
      level.sublevels &&
      sublevelIndex < level.sublevels.length
    ) {
      dispatch(setCurrentLevelId(level.sublevels[sublevelIndex].id));
    } else {
      dispatch(setCurrentLevelId(level.id));
    }
```

Reply:
Personally, I think that the current implementation may be easier to read. For me, it compartmentalizes the assignment of `newLevelId` and then what is done with the variable.

Reply:
For me I see a `let` and alarm bells just start going off in my head. If a mutable variable is needed, something complicated/delicate/error-prone is likely to be coming up next, and I need to be extra careful when reading or touching that code. 

I do see your point about giving `newLevelId` a name and that improves readability. How about this? 
```suggestion
    if (
      sublevelIndex !== undefined &&
      level.sublevels &&
      sublevelIndex < level.sublevels.length
    ) {
      const newLevelId = level.sublevels[sublevelIndex].id;
      dispatch(setCurrentLevelId(newLevelId));
    } else {
      const newLevelId = level.id;
      dispatch(setCurrentLevelId(newLevelId));
    }

```

Reply:
Nice! And thanks for taking time looking at a student-labs PR. Appreciate getting different perspectives.

Reply:
Looks fine to me, will update!